### DISCUSSION TITLE
Is backup and restore using borgbackup working properly?

### DISCUSSION THREAD

**Discussion by AwesomeStickz** [WeblateOrg/weblate#8009] [Q&A]
I'm trying to move weblate from one server to another server and everything seems to be fine except the moving of the database, it doesn't go to the same state it was in when I restore the db with `psql --file=database.sql weblate -h <containerId>`, I'm not sure if I'm doing it wrong or not, but this is what I'm doing:

![image](https://user-images.githubusercontent.com/38146668/184961419-89b9ea07-0a14-452b-9493-c93544a9e178.png)

And it does seem to copy some things but I don't see any projects there (am I supposed to create the project by myself and can't let the db backup create it instead?) and no info about my weblate admin profile and other stuff, I even doubt if it did it properly or not so here are a few screenshots of the output when I run that command:

![image](https://user-images.githubusercontent.com/38146668/184962038-1922791e-c0c6-459a-9e0e-71f9191c9d73.png)
![image](https://user-images.githubusercontent.com/38146668/184962045-af79b55b-3a3d-4ba9-8b04-62b8dd8db677.png)
![image](https://user-images.githubusercontent.com/38146668/184962055-a56045b5-4978-4867-9042-cec34977d760.png)
![image](https://user-images.githubusercontent.com/38146668/184962068-32bf56c6-a434-4ee1-8cb4-015c53f9eb46.png)
![image](https://user-images.githubusercontent.com/38146668/184962077-8f59f2e5-d236-4e10-9fba-0dbef7f94db6.png)
![image](https://user-images.githubusercontent.com/38146668/184962085-76c32230-830a-48c5-8d24-398c0c8df386.png)

I'm not sure if these errors prevents the data to be restored but if that's the case, what am I even supposed to do? Because it's the default new installation of weblate and restoring using that, no idea what's the problem here, any idea what could be wrong?

**Comment by nijel** [comment]
It seems that you didn't start with a blank database when restoring, but it already contained some tables.

**Reply by AwesomeStickz** [reply]
Which I'm not sure how to, because it was just a fresh installation of weblate in new server, and then started it with `docker compose up -d` and then I did this backup restore stuff, anything I'm missing probably?

**Reply by nijel** [reply]
The Docker container automatically creates the database upon start.

**Reply by AwesomeStickz** [reply]
So here's the things I did this time:

1. `docker exec -it <weblateContainerId> /bin/bash`
2. `psql postgres -h <psqlContainerId>`
3. `DROP DATABASE weblate;`
4. `CREATE DATABASE weblate;`
5. `\q` to exit from psql shell
6. `cd app/data/backups` to get into the backups folder where the borg backup sql file is present
7. `psql --file=database.sql weblate -h <psqlContainerId>`

It ran all the commands and all alter table and other queries ran perfectly with no errors like before, but when I tried to check weblate in the website, it still didn't have all the data in it, and then I restarted all containers to see if it was because of redis cache and that wasn't the case either

**Reply by AwesomeStickz** [reply]
Not sure if this plays a role here but I didn't make use of these files:

![image](https://user-images.githubusercontent.com/38146668/185090360-c10f078e-a9c2-4d2a-9e67-0fb150f03cfa.png)

I used the same `docker-compose.override.yml` file I used in my old server and thought that's enough, maybe not using the above 3 files could be a problem?

**Reply by AwesomeStickz** [reply]
Actually nvm, it somehow had another database.sql file which had like just 2MB data, which explains it having no data above when I tried to restore from backup but yes, now it works. After the step 7 I mentioned above, I had to kill and start the containers again, wait a minute or two for the migration that weblate does and then it shows the latest data. So I assume it works now, thanks for letting me know that I had to start with an empty database without letting weblate create the db again, otherwise I was assuming that weblate wouldn't interfere with the database and was thinking that the backup should've restored properly

**Reply by nijel** [reply]
If you restart the container, it will again filll the database, so you need to again DROP and CREATE it.

**Reply by AwesomeStickz** [reply]
No, I mean like this: DELETE DATABASE, CREATE DATABASE, `psql --file=database.sql weblate -h <psqlContainerId>` to copy the data from SQL file, and then restart container (because if you don't restart after restoring in the above way, it says internal server error).

And that worked as I said
